### DISCUSSION TITLE
Osconfig in nested environments

### DISCUSSION THREAD

**Discussion by alziegel** [Azure/azure-osconfig#451] [General]
Hi, is it possible to use osconfig with iotedge nesting, one acting as an internet connected gateway for other non internetconnected iotedge devices?

thanks in advance

**Comment by MatthewMWR** [comment]
Generally, yes. We have successfully tested running osconfig on a child device, communicating to IoT Hub indirectly via an iot edge parent device. My memory is that you configure the child device as if the parent device were the IoT Hub. It has been some time since I tried this, so the details are not fresh in my mind. 

Although you didn't ask about proxy servers, I'll mention anyway (since these often come up in layered environments) that alternatively osconfig can reach the IoT Hub via a proxy server.

Let me know if you need more information. We might take the opportunity to make sure our docs are clear and current for this scenario.

The following doc section mentions both the proxy and the parent/child scenarios https://learn.microsoft.com/azure/osconfig/howto-install?tabs=package#optional-agent-settings
